casepro/statistics/models.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import unicode_literals
+
+import pytz
+import six
+
+from dash.orgs.models import Org
+from django.contrib.auth.models import User
+from django.core.cache import cache
+from django.db import models, connection
+from django.db.models import Sum
+from django.utils.functional import SimpleLazyObject
+from django.utils.translation import ugettext_lazy as _
+
+from casepro.cases.models import Partner
+
+
+class DailyCount(models.Model):
```

Comment:
So despite previous comments, I've gone with a single count model, and ditched all the foreign keys in favour of single scope field which has all that encoded. I was concerned about the performance of that, but there's an index on (item_type, scope, day) and it all seems fine.

The typical usage pattern as you'll see in the tests is to call a method to return a set of counts, and then call another method on that to aggregate it in the way you need, e.g.

```python
DailyCount.get_by_partner([partner1], DailyCount.TYPE_REPLIES).total()
DailyCount.get_by_org([org1, org2], DailyCount.TYPE_REPLIES).day_totals()
```